register_cc_darwin.go

```diff
@@ -0,0 +1,9 @@
+//go:build (arm64 && go1.18) || (amd64 && go1.17)
```

Comment:
This is just my preference, but could you rename the file to `registercc_darwin.go`?